examples/fast_neural_style/neural_style.py

```diff
@@ -0,0 +1,249 @@
+        style = Image.fromarray(imarray)
+
+    style = style_transform(style)
+
+    style = style.repeat(args.batch_size, 1, 1, 1).to(device)
+
+    features_style = vgg(utils.normalize_batch(style))
+    gram_style = [utils.gram_matrix(y) for y in features_style]
+
+    running_avgs = OrderedDict()
+
+    def step(engine, batch):
+
+        x, _ = batch
+        x = x.to(device)
+
+        n_batch = len(x)
+
+        transformer.zero_grad()
+
+        y = transformer(x)
+
+        x = utils.normalize_batch(x)
+        y = utils.normalize_batch(y)
+
+        features_x = vgg(x)
+        features_y = vgg(y)
+
+        content_loss = args.content_weight * mse_loss(features_y.relu2_2, features_x.relu2_2)
+
+        style_loss = 0.
+        for ft_y, gm_s in zip(features_y, gram_style):
+            gm_y = utils.gram_matrix(ft_y)
+            style_loss += mse_loss(gm_y, gm_s[:n_batch, :, :])
+        style_loss *= args.style_weight
+
+        total_loss = content_loss + style_loss
+        total_loss.backward()
+        optimizer.step()
+
+        return {
+            'content_loss': content_loss.item(),
+            'style_loss': style_loss.item(),
+            'total_loss': total_loss.item()
+        }
+
+    trainer = Engine(step)
+    checkpoint_handler = ModelCheckpoint(args.checkpoint_model_dir, 'checkpoint',
+                                         save_interval=args.checkpoint_interval,
+                                         n_saved=10, require_empty=False, create_dir=True)
+
+    progress_bar = Progbar(loader=train_loader, metrics=running_avgs)
+
+    @trainer.on(Events.EPOCH_COMPLETED)
+    def stylize_image(engine):
+        path = os.path.join(args.stylized_test_dir, STYLIZED_IMG_FNAME.format(engine.state.epoch))
+
+        if args.test_image != 'test':
+            content_image = utils.load_image(args.test_image, scale=None)
+        else:
+            imarray = np.random.rand(32, 32, 3).astype('uint8')
+            content_image = Image.fromarray(imarray)
+
+        content_transform = transforms.Compose([
+            transforms.ToTensor(),
+            transforms.Lambda(lambda x: x.mul(255))
+        ])
+        content_image = content_transform(content_image)
+        content_image = content_image.unsqueeze(0).to(device)
+
+        with torch.no_grad():
+            output = transformer(content_image).cpu()
+
+        utils.save_image(path, output[0])
+
+    trainer.add_event_handler(event_name=Events.EPOCH_COMPLETED, handler=checkpoint_handler,
+                              to_save={'net': transformer})
+    trainer.add_event_handler(event_name=Events.ITERATION_COMPLETED, handler=progress_bar)
+    trainer.run(train_loader, max_epochs=args.epochs)
+
+
+def stylize(args):
+    device = torch.device("cuda" if args.cuda else "cpu")
+
+    if args.content_image != 'test':
+        content_image = utils.load_image(args.content_image, scale=args.content_scale)
+    else:
```

Comment:
we should remove this testing code from here. We don't actually need to do the smoke test and see results

Reply:
Incorporated